Sources/ClientRuntime/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -84,30 +84,56 @@ public class CRTClientEngine: HttpClientEngine {
         let connectionMgr = try await serialExecutor.getOrCreateConnectionPool(endpoint: request.endpoint)
         let connection = try await connectionMgr.acquireConnection()
         self.logger.debug("Connection was acquired to: \(String(describing: request.endpoint.url?.absoluteString))")
-        return try await withCheckedThrowingContinuation({ (continuation: StreamContinuation) in
-            do {
-                let requestOptions = try makeHttpRequestStreamOptions(request, continuation)
-                let stream = try connection.makeRequest(requestOptions: requestOptions)
-                try stream.activate()
-            } catch {
-                continuation.resume(throwing: error)
+        switch connection.httpVersion {
+        case .version_1_1:
+            self.logger.debug("Using HTTP/1.1 connection")
+            let crtRequest = try request.toHttpRequest()
+            return try await withCheckedThrowingContinuation { (continuation: StreamContinuation) in
+                let requestOptions = makeHttpRequestStreamOptions(request: crtRequest,
+                                                                  continuation: continuation)
+                do {
+                    let stream = try connection.makeRequest(requestOptions: requestOptions)
+                    try stream.activate()
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+        case .version_2:
+            self.logger.debug("Using HTTP/2 connection")
+            let crtRequest = try request.toHttp2Request()
+
+            return try await withCheckedThrowingContinuation { (continuation: StreamContinuation) in
+                let requestOptions = makeHttpRequestStreamOptions(request: crtRequest,
+                                                                  continuation: continuation,
+                                                                  http2ManualDataWrites: true)
+                Task {
+                    let stream = try connection.makeRequest(requestOptions: requestOptions) as! HTTP2Stream
+                    try stream.activate()
+                    do {
+                        try await stream.write(body: request.body)
+                    } catch {
+                        self.logger.error(error.localizedDescription)
+                    }
+                }
             }
-        })
+        case .unknown:
+            fatalError()
+        }
     }
 
-    public func makeHttpRequestStreamOptions(
-        _ request: SdkHttpRequest,
-        _ continuation: StreamContinuation
-    ) throws -> HTTPRequestOptions {
+    private func makeHttpRequestStreamOptions(
```

Comment:
add docs

Reply:
Just a reminded to make sure to describe `http2ManualDataWrites`, what it does and why its needed

Reply:
updated.